core/java/com/google/instrumentation/stats/AggregationDescriptor.java

```diff
@@ -0,0 +1,148 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.Duration;
+import com.google.instrumentation.common.MatchFunction;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * An {@link AggregationDescriptor} describes an Aggregation, which is a series of individual
+ * measurements.
+ */
+public abstract class AggregationDescriptor {
+  /**
+   * Applies the given match function to the underlying data type.
+   */
+  public abstract <T> T match(
+      MatchFunction<DistributionAggregationDescriptor, T> p0,
+      MatchFunction<IntervalAggregationDescriptor, T> p1);
+
+  // Seals the disjoint union.
+  private AggregationDescriptor() {
+  }
+
+  /**
+   * An {@link AggregationDescriptor} based on distributions.
+   *
+   * <p>A distribution aggregation may optionally contain a histogram of the values in the
+   * population. The bucket boundaries for that histogram are described by
+   * {@link getBucketBoundaries}, which defines {@code getBucketBoundaries.size() + 1 (= N)}
+   * buckets. The boundaries for bucket index i are:
+   * <ul>
+   * <li>[-infinity, bounds[i]) for i == 0
+   * <li>[bounds[i-1], bounds[i]) for 0 &lt; i &lt; N-2
+   * <li>[bounds[i-1], +infinity) for i == N-1
+   * </ul>
+   * i.e. an underflow bucket (number 0), zero or more finite buckets (1 through N - 2, and an
+   * overflow bucket (N - 1), with inclusive lower bounds and exclusive upper bounds.
+   *
+   * <p>Note: If N = 1, there are no finite buckets and the single bucket is both the overflow
+   * and underflow bucket.
+   */
+  public static final class DistributionAggregationDescriptor extends AggregationDescriptor {
+    /**
+     * Constructs a new {@link DistributionAggregationDescriptor}.
+     */
+    public static DistributionAggregationDescriptor create(List<Double> bucketBoundaries) {
+      return new DistributionAggregationDescriptor(
+          Collections.unmodifiableList(new ArrayList<Double>(bucketBoundaries)));
+    }
+
+    /**
+     * Constructs a new {@link DistributionAggregationDescriptor} without the optional
+     * histogram bucket boundaries.
+     */
+    public static DistributionAggregationDescriptor create() {
+      return new DistributionAggregationDescriptor(null);
+    }
+
+    /**
+     * The (possibly null) histogram bucket boundaries for a distribution.
+     *
+     * <p>Note: The returned list is unmodifiable, attempts to update it will throw an
+     * UnsupportedOperationException.
+     */
+    @Nullable
+    public List<Double> getBucketBoundaries() {
```

Comment:
The histogram bucket boundaries are optional, which is described in the class description and  factory methods. I've updated the comment to make it more clear here.